spring-cloud-gcp-pubsub/src/main/java/org/springframework/cloud/gcp/pubsub/core/subscriber/PubSubSubscriberOperations.java

```diff
@@ -47,6 +49,16 @@
 	 */
 	Subscriber subscribe(String subscription, MessageReceiver messageReceiver);
 
+	/**
+	 * Add a callback method to an existing subscription.
+	 * <p>The created {@link Subscriber} is returned so it can be stopped.
+	 * @param subscription the name of an existing subscription
+	 * @param messageConsumer the callback method triggered when new messages arrive
+	 * @return subscriber listening to new messages
+	 * @since 1.1
+	 */
+	Subscriber subscribe(String subscription, Consumer<BasicAcknowledgeablePubsubMessage> messageConsumer);
```

Comment:
I guess we need an implementation for this method.
Why then do we need a `MessageReceiver`-based API then?
Aren't you going to hide such a detail behind our own API?

Reply:
The implementation is coming. I just wanted to get some preliminary feedback on the idea.
The two reasons I'm thinking of keeping the `MessageReceiver` API are (1) it's already in `RC1` and people are using it, and (2) it's the low-level API that Pub/Sub provides.
(1) is my main concern because I don't think we can have a breaking change like removing that method in `1.1`. WDYT?

Reply:
We can deprecate it instead, although I'm fine to keep it for time being.

The API idea is OK I believe...

Reply:
Leave the method and deprecate it would be developer friendly.

Reply:
Sounds good.